### DISCUSSION TITLE
Is there a Python API to invoke kypher queries?

### DISCUSSION THREAD

**Discussion by szeke** [usc-isi-i2/kgtk#413] [Q&A] [answered]
I'd like to run a templetized kypher query many times, and the overhead of invoking KGTK is high. Can I invoke kypher directly from python? For example, I want to run the following query many times, substituting the q-node by different nodes. 
```
kgtk query -i "$OUT"/coauthors.context.tsv.gz \
--match 'coauthors: (:Q100057080)-[]->(context)' \
--return 'context' 
```

Thanks

**Comment by chalypso** [comment, marked as answer]
I am working on one, it's close to done, I should have a first version checked in by the beginning of the week.

**Reply by chalypso** [reply]
See https://github.com/usc-isi-i2/kgtk/blob/dev/kgtk/kypher/api.py
